packages/commonwealth/server/workers/messageRelayer/relayForever.ts

```diff
@@ -42,7 +42,8 @@ export async function relayForever(maxIterations?: number) {
       stats().gauge('messageRelayerNumUnrelayedEvents', numUnrelayedEvents);
     }
 
-    if (numUnrelayedEvents === 0) {
+    // TODO: Figure out why numUnrelayedEvents is negative. Skipping sleep step causing event queue to clog
+    if (numUnrelayedEvents <= 0) {
```

Comment:
No need to change this.

A negative value likely means that the trigger is implemented wrong. If you copied and pasted from the migration, then check for a later migration that fixed the PG trigger.

Reply:
Its necessary to change becuase if it is less than 0 then this will keep running without sleeping, starving the other async tasks.

I added the PG trigger directly from the migration, without it, this won't increment/decrement at all and the message will not be sent to the outbox.

Reply:
There are 2 migrations for the pg trigger– one to create and one to fix it. Your code resembles the initial migration which was broken and caused a similar bug to what you're experiencing.

Check the `fix-pg-notify-triggger.js` migration– it's slightly updated.

Reply:
Yeah ok the fix migration seems to have fixed it. I have reverted this line, thanks!